### PR TITLE
Add Feather RP2040 DVI (plz read)

### PR DESCRIPTION
**Pull request by PaintYourDragon**
Prob don’t want to merge this as-is, it’s an experiment as I learn the ropes.

There will be a number of incoming boards based on a common RP2040 design…very similar to the SCORPIO design, with an extra eight consecutive GPIO connected to special functions. Most of these should be very simple to add here: copy the adafruit_feather_rp2040_scorpio directory to a new name, then edit the files within to assign this board a unique USB PID and to provide names for the extra GPIOs (e.g. CKN, CKP, etc. in this one). So initially this is just a test to see that I’ve got that aspect right.

Later…the Feather RP2040 DVI will be an odd one in that it’s overclocked (TBD whether this happens on startup or doesn’t switch over until user code specifically enables DVI out)…but also, at very high overclock settings, may need to use QSPI clock div4 instead of div2. No simple settings for these things in the aforementioned files, presumably this will involve digging deeper into CircuitPython source. The USB host Feather might also involve some overclocking.

So, just need to know, do the changes here look about right for one of these future Feathers that just need a name and PID and then names assigned to special pins? None of the weird overclock stuff, just special I/O. Thx.

### CODE REVIEW

**Comment by dglaude** [comment]
I am not an expert... but I believe the solderparty_rp2040_stamp is using a mechanism to have multiple "personalities" depending on the daughter board the stamp reside on.
Basically, there is a little python file you import to have the proper pin mapping, and that file is merged into the firmware.
I found that a "sexy" way to avoid having different firmware for different variation of the same thing (RP2040).
But there are argument like: "GitHub provide the Action for free" and "It is easier for the end-user to have a dedicated UF2 for their specific board".

PS: I am interested in how the RP2040 bones(?) derived board will be integrated... I really want the DVI and the USB Host one (I already have a Scorpio).

File: ports/raspberrypi/boards/adafruit_feather_rp2040_dvi/pins.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    { MP_ROM_QSTR(MP_QSTR_A3), MP_ROM_PTR(&pin_GPIO29) },
+    { MP_ROM_QSTR(MP_QSTR_D24), MP_ROM_PTR(&pin_GPIO24) },
+    { MP_ROM_QSTR(MP_QSTR_D25), MP_ROM_PTR(&pin_GPIO25) },
+    { MP_ROM_QSTR(MP_QSTR_SCK), MP_ROM_PTR(&pin_GPIO14) },
+    { MP_ROM_QSTR(MP_QSTR_MOSI), MP_ROM_PTR(&pin_GPIO15) },
+    { MP_ROM_QSTR(MP_QSTR_MISO), MP_ROM_PTR(&pin_GPIO8) },
+    { MP_ROM_QSTR(MP_QSTR_D0), MP_ROM_PTR(&pin_GPIO1) },
+    { MP_ROM_QSTR(MP_QSTR_RX), MP_ROM_PTR(&pin_GPIO1) },
+    { MP_ROM_QSTR(MP_QSTR_D1), MP_ROM_PTR(&pin_GPIO0) },
+    { MP_ROM_QSTR(MP_QSTR_TX), MP_ROM_PTR(&pin_GPIO0) },
```
Comment (dhalbert) [inline ports/raspberrypi/boards/adafruit_feather_rp2040_dvi/pins.c:18]: Aliased pins have blank lines surrounding, for clarity.
```suggestion

    { MP_ROM_QSTR(MP_QSTR_D0), MP_ROM_PTR(&pin_GPIO1) },
    { MP_ROM_QSTR(MP_QSTR_RX), MP_ROM_PTR(&pin_GPIO1) },
    
    { MP_ROM_QSTR(MP_QSTR_D1), MP_ROM_PTR(&pin_GPIO0) },
    { MP_ROM_QSTR(MP_QSTR_TX), MP_ROM_PTR(&pin_GPIO0) },
    
```

File: ports/raspberrypi/boards/adafruit_feather_rp2040_dvi/pins.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    { MP_ROM_QSTR(MP_QSTR_SDA), MP_ROM_PTR(&pin_GPIO2) },
+    { MP_ROM_QSTR(MP_QSTR_SCL), MP_ROM_PTR(&pin_GPIO3) },
+    { MP_ROM_QSTR(MP_QSTR_D5), MP_ROM_PTR(&pin_GPIO5) },
+    { MP_ROM_QSTR(MP_QSTR_D6), MP_ROM_PTR(&pin_GPIO6) },
+    { MP_ROM_QSTR(MP_QSTR_D9), MP_ROM_PTR(&pin_GPIO9) },
+    { MP_ROM_QSTR(MP_QSTR_D10), MP_ROM_PTR(&pin_GPIO10) },
+    { MP_ROM_QSTR(MP_QSTR_D11), MP_ROM_PTR(&pin_GPIO11) },
+    { MP_ROM_QSTR(MP_QSTR_D12), MP_ROM_PTR(&pin_GPIO12) },
+    { MP_ROM_QSTR(MP_QSTR_D13), MP_ROM_PTR(&pin_GPIO13) },
+    { MP_ROM_QSTR(MP_QSTR_LED), MP_ROM_PTR(&pin_GPIO13) },
```
Comment (dhalbert) [inline ports/raspberrypi/boards/adafruit_feather_rp2040_dvi/pins.c:31]:
```suggestion

    { MP_ROM_QSTR(MP_QSTR_D13), MP_ROM_PTR(&pin_GPIO13) },
    { MP_ROM_QSTR(MP_QSTR_LED), MP_ROM_PTR(&pin_GPIO13) },
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Looks fine! I added some blank lines for clarity.

**Comment by dhalbert** [comment]
> So, just need to know, do the changes here look about right for one of these future Feathers that just need a name and PID and then names assigned to special pins? None of the weird overclock stuff, just special I/O. Thx.

I think so.

**Comment by dhalbert** [comment]
@PaintYourDragon Note that it's better to create a branch for your PR rather than edit your `main` branch. If you do the latter then it can get out of sync with the upstream `main` branch. See https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github/always-work-on-a-branch#create-your-new-branch-2992239

**Pull request closed without merging**